content/ngf/overview/gateway-architecture.md

```diff
@@ -24,87 +24,331 @@ NGINX Gateway Fabric is an open source project that provides an implementation o
+        AppDevB[Application Developer B]
+        AppDevC[Application Developer C]
+    end
+
+    subgraph Clients[Clients]
+        ClientsA[Clients A]
+        ClientsB[Clients B]
+        ClientsC[Clients C]
+    end
+
+    subgraph "Public Endpoints"
+        PublicEndpointAB[Public Endpoint AB<br>TCP Load Balancer/NodePort]
+        PublicEndpointC[Public Endpoint C<br>TCP Load Balancer/NodePort]
+    end
+
+    %% Updated Traffic Flow
+    ClientsA == a.example.com ==> PublicEndpointAB
+    ClientsB == b.example.com ==> PublicEndpointAB
+    ClientsC == c.other-example.com ==> PublicEndpointC
+
+    PublicEndpointAB ==> NGINXProcessAB
+    PublicEndpointC ==> NGINXProcessC
+    NGINXProcessAB ==> ApplicationA
+    NGINXProcessAB ==> ApplicationB
+    NGINXProcessC ==> ApplicationC
+
+    %% Kubernetes Configuration Flow
+    HTTPRouteA --> GatewayAB
+    HTTPRouteB --> GatewayAB 
+    HTTPRouteC --> GatewayC
+
+    NGFPod --> KubernetesAPI
+    NGFPod --gRPC--> NGINXAgentAB
+    NGINXAgentAB --> NGINXProcessAB
+    NGFPod --gRPC--> NGINXAgentC
+    NGINXAgentC --> NGINXProcessC
+
+    ClusterOperator --> KubernetesAPI
+    AppDevA --> KubernetesAPI
+    AppDevB --> KubernetesAPI
+    AppDevC --> KubernetesAPI
+
+    %% Styling
+    style ClusterOperator fill:#66CDAA,stroke:#333,stroke-width:2px
+    style GatewayAB fill:#66CDAA,stroke:#333,stroke-width:2px
+    style GatewayC fill:#66CDAA,stroke:#333,stroke-width:2px
+    style NGFPod fill:#66CDAA,stroke:#333,stroke-width:2px
+
+    style NGINXProcessAB fill:#66CDAA,stroke:#333,stroke-width:2px
+    style NGINXProcessC fill:#66CDAA,stroke:#333,stroke-width:2px
+
+    style KubernetesAPI fill:#8A2BE2,stroke:#333,stroke-width:2px
+
+    style HTTPRouteAAndApplicationA fill:#E0FFFF,stroke:#333,stroke-width:2px
+    style HTTPRouteBAndApplicationB fill:#E0FFFF,stroke:#333,stroke-width:2px
+
+    style AppDevA fill:#FFA07A,stroke:#333,stroke-width:2px
+    style HTTPRouteA fill:#FFA07A,stroke:#333,stroke-width:2px
+    style ApplicationA fill:#FFA07A,stroke:#333,stroke-width:2px
+    style ClientsA fill:#FFA07A,stroke:#333,stroke-width:2px
+
+    style AppDevB fill:#87CEEB,stroke:#333,stroke-width:2px
+    style HTTPRouteB fill:#87CEEB,stroke:#333,stroke-width:2px
+    style ApplicationB fill:#87CEEB,stroke:#333,stroke-width:2px
+    style ClientsB fill:#87CEEB,stroke:#333,stroke-width:2px
+
+    style AppDevC fill:#FFC0CB,stroke:#333,stroke-width:2px
+    style HTTPRouteC fill:#FFC0CB,stroke:#333,stroke-width:2px
+    style ApplicationC fill:#FFC0CB,stroke:#333,stroke-width:2px
+    style ClientsC fill:#FFC0CB,stroke:#333,stroke-width:2px
+
+    style PublicEndpointAB fill:#FFD700,stroke:#333,stroke-width:2px
+    style PublicEndpointC fill:#FFD700,stroke:#333,stroke-width:2px
+
+    %% Styling
+    classDef dashedSubgraph stroke-dasharray: 5, 5;
+
+    %% Assign Custom Style Classes
+    class DataplaneComponentsAB dashedSubgraph;
+    class DataplaneComponentsC dashedSubgraph;
+```
```

Comment:
The only downside of using mermaid versus manual is it doesn't render the prettiest or most balanced diagrams. (like the giant nginx-gateway namespace box). Unless you can create a manual image and translate into mermaid?

Reply:
I'll see if I can tweak it a bit more, but yeah, it is an issue.

Reply:
Yeah, that does look a bit better. Only nitpick would be the AB components are below the C components.

Reply:
I'll see if I can reorder them